system/ee/ExpressionEngine/Cli/Commands/CommandGenerateTemplates.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * This source file is part of the open source project
+ * ExpressionEngine (https://expressionengine.com)
+ *
+ * @link      https://expressionengine.com/
+ * @copyright Copyright (c) 2003-2023, Packet Tide, LLC (https://www.packettide.com)
+ * @license   https://expressionengine.com/license Licensed under Apache License, Version 2.0
+ */
+
+namespace ExpressionEngine\Cli\Commands;
+
+use ExpressionEngine\Cli\Cli;
+
+/**
+ * Command to make action files for addons
+ */
+class CommandGenerateTemplates extends Cli
+{
+    /**
+     * name of command
+     * @var string
+     */
+    public $name = 'Generate Templates';
+
+    /**
+     * signature of command
+     * @var string
+     */
+    public $signature = 'generate:templates';
+
+    /**
+     * How to use command
+     * @var string
+     */
+    public $usage = 'php eecli.php generate:templates [generator] [--options]';
+
+    /**
+     * options available for use in command
+     * @var array
+     */
+    public $commandOptions = [
+        'list,l'        => 'command_generate_templates_list_generators',
+        /* 'themes,t'      => 'command_generate_templates_list_themes', */
+        'show,s'        => 'command_generate_templates_show_template_content'
+    ];
+
+    protected $data = [];
+
+    /**
+     * Run the command
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Get all the template generators
+        $generatorsList = ee('TemplateGenerator')->registerAllTemplateGenerators();
+
+        // do we need to just list all possible generators?
+        if ($this->option('--list', false)) {
+            $this->info('command_generate_templates_listing_generators');
+            $this->data['table'] = [];
+            array_walk($generatorsList, function ($generator, $key) {
+                $name = $generator->getName();
+                if (!empty($name)) {
+                    $this->data['table'][] = [$key, $name];
+                }
+            });
+            $this->table([
+                lang('name'),
+                lang('description'),
+            ], $this->data['table']);
+            $this->complete();
+        }
+
+        // list the themes that are additionally available
+        /* if ($this->option('--themes', false)) {
```

Comment:
I left all the themes work commented out for now because I think there's a good chance we may use some of the code later.  My vote would be to keep it for now